test/ServiceTest.cpp

```diff
@@ -212,20 +211,38 @@ TEST_F(ServiceTest, computeResult) {
   // CHECK 1: An exception shall be thrown, when
   // status-code isn't ok, contentType doesn't match
   expectThrowOrSilence(
-      "", "SERVICE responded with HTTP status code: 400, Bad Request.",
+      genJsonResult({"x", "y"}, {{"bla", "bli"}, {"blu"}, {"bli", "blu"}}),
+      "SERVICE responded with HTTP status code: 400, Bad Request.",
       boost::beast::http::status::bad_request,
       "application/sparql-results+json");
 
-  expectThrowOrSilence("",
-                       "QLever requires the endpoint of a SERVICE to send "
-                       "the result as 'application/sparql-results+json' but "
-                       "the endpoint sent 'wrong/type'.",
-                       boost::beast::http::status::ok, "wrong/type");
+  expectThrowOrSilence(
+      genJsonResult({"x", "y"}, {{"bla", "bli"}, {"blu"}, {"bli", "blu"}}),
+      "QLever requires the endpoint of a SERVICE to send "
+      "the result as 'application/sparql-results+json' but "
+      "the endpoint sent 'wrong/type'.",
+      boost::beast::http::status::ok, "wrong/type");
 
   // or Result has invalid structure
   expectThrowOrSilence("{\"invalid\": \"structure\"}",
                        "JSON result does not have the expected structure.");
 
+  expectThrowOrSilence("", "JSON result does not have the expected structure.");
+
+  // `head`/`vars` missing
+  expectThrowOrSilence(
+      "{\"results\": {\"bindings\": {{\"x\": {{\"type\": \"uri\"}, {\"value\": "
+      "\"a\"}}, \"y\": {{\"type\": \"uri\"}, {\"value\": \"b\"}}}}}}",
+      "JSON result does not have the expected structure.");
+  expectThrowOrSilence(
+      "{\"head\": {},"
+      "\"results\": {\"bindings\": {}}}",
+      "JSON result does not have the expected structure.");
+  // wrong variables type (array of strings expected)
+  expectThrowOrSilence(
+      "{\"head\": {\"vars\": 1},"
+      "\"results\": {\"bindings\": {}}}",
+      "JSON result does not have the expected structure.");
```

Comment:
I think a test case for `results section missing` and `results section has the wrong structure` is missing.
In particular we probably want to catch the internal exceptions of the LazyJson parser and make something user-friendly out of them. (People will complain to us, even if its the fault of the other endpoint, so we should help our future selves as much as possible.